### DISCUSSION TITLE
git-push access denied if done from a codespace created from a pull request.

### DISCUSSION THREAD

**Discussion by eps1lon** [community/community#36299] [Codespaces]
As an admin on a repository, I created a codespace from a pull request (here: https://github.com/typescript-cheatsheets/react/pull/573).

After I made changes in the pull request branch, I tried to push the changes. However, a simple `git push` did not work because the created branch was not tracking a remote branch. But even a `git push upstream patch-6` got access denied:

```bash
$ git push upstream patch-6
remote: Permission to MananTank/react.git denied to eps1lon.
fatal: unable to access 'https://github.com/MananTank/react/': The requested URL returned error: 403
```

In other words, creating a codespace from a pull request opens the branch in read-only mode. This is inconsistent with the `gh` CLI.

Note that this is not caused by the PR author not allowing edits to PRs by maintainers.

I did a `gh pr checkout 573` locally and was able to push changes (as expected) with a simple `git push`:

```bash
git push
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 6 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (3/3), 1.77 KiB | 907.00 KiB/s, done.
Total 3 (delta 2), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To https://github.com/MananTank/react.git
   62e7f3f..4f38baf  patch-6 -> patch-6
```

This makes codespaces in their current form not suitable for amending PRs as maintainers of OSS repositories.

**Comment by cwndrws** [comment]
Thanks for reporting the issue. This seems to be a recent regression and we're tracking the issue internally.
